plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -61,610 +64,934 @@ describe('PersonState.update()', () => {
+                    uuid: uuid2.toString(),
+                    properties: {},
+                    created_at: timestamp2,
+                    version: 1,
+                    is_identified: true, // TODO: we don't want this person to be updated, but the current code does so
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['old-user']))
+            const distinctIds2 = await hub.db.fetchDistinctIdValues(persons[1])
+            expect(distinctIds2).toEqual(expect.arrayContaining(['new-user']))
+
+            // verify personContainer
+            expect(persons[1]).toEqual(await personContainer.get())
+        })
+
+        it('does not merge people when both users are identified', async () => {
+            await hub.db.createPerson(timestamp, {}, {}, {}, 2, null, true, uuid.toString(), ['old-user'])
+            await hub.db.createPerson(timestamp2, {}, {}, {}, 2, null, true, uuid2.toString(), ['new-user'])
+
+            const personContainer = await personState({
+                event: '$identify',
+                distinct_id: 'new-user',
+                properties: {
+                    $anon_distinct_id: 'old-user',
+                },
+            }).update()
 
-        expect(hub.db.fetchPerson).toHaveBeenCalledTimes(1) // It does a single reset after failing once
-        expect(hub.personManager.isNewPerson).toHaveBeenCalledTimes(0)
+            // verify Postgres persons
+            const persons = (await hub.db.fetchPersons()).sort((a, b) => a.id - b.id)
+            expect(persons.length).toEqual(2)
+            expect(persons[0]).toEqual(
+                expect.objectContaining({
+                    id: expect.any(Number),
+                    uuid: uuid.toString(),
+                    properties: {},
+                    created_at: timestamp,
+                    version: 0,
+                    is_identified: true,
+                })
+            )
+            expect(persons[1]).toEqual(
+                expect.objectContaining({
+                    id: expect.any(Number),
+                    uuid: uuid2.toString(),
+                    properties: {},
+                    created_at: timestamp2,
+                    version: 0,
+                    is_identified: true,
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['old-user']))
+            const distinctIds2 = await hub.db.fetchDistinctIdValues(persons[1])
+            expect(distinctIds2).toEqual(expect.arrayContaining(['new-user']))
+
+            // verify personContainer
+            expect(persons[1]).toEqual(await personContainer.get())
+        })
+
+        describe('with ClickHouse', () => {
```

Comment:
This separation is unhealthy for test coverage (we need to repeat ourselves) and probably unneeded, but also really hard to undo. :/